src/python/pants/backend/project_info/dependencies.py

```diff
@@ -43,7 +44,11 @@ async def dependencies(
 ) -> Dependencies:
     if dependencies_subsystem.transitive:
         transitive_targets = await Get(
-            TransitiveTargets, TransitiveTargetsRequest(addresses, include_special_cased_deps=True)
+            TransitiveTargets,
+            TransitiveTargetsRequest(
+                addresses,
+                should_resolve_deps_predicate=should_resolve_all_deps_predicate,
+            ),
```

Comment:
@stuhood :tada: This is what it looks like to replace the `include_special_cased_deps` flag with a predicate that says "always include all deps"